models/order.js

```diff
@@ -12,7 +13,68 @@ const Entry = new Schema({
 const orderSchema = new Schema({
   id: String,
   status: { type: String, default: constants.OPEN },
-  items: [Entry]
+  entries: [Entry]
 });
 
+orderSchema.statics.getOrders = function() {
+  return getOrders.apply(this)
+  .then(getEntries)
+  .catch((e) => {
+    throw new Error(e);
+  });
+}
+
+function getOrders() {
+  const Order = this;
+
+  return new Promise((resolve, reject) => {
+    Order.find().exec((err, orders) => {
+      if(err) {
+        reject(err);
+      }
+
+      resolve(orders.map(toOrder));
+    });
+  });
+}
+
+function populateEntries(order) {
+  return new Promise((resolve, reject) => {
+    Item.populate(order.entries, [{ path: 'item_id', model: 'Item', select: 'name price -_id' }], (err, res) => {
```

Comment:
This essentially replaces each `order.entries.item_id` with an actual `Item` document right? Since we already have the `order` and its associated `entries` here, would it work if we just do the assignment / mapping `toEntry` here and resolve with the whole `order`, instead of iterating over `orders` again and messing with `index` below? We would eliminate the extra iteration and `getEntries` could then just be simple:

`return Promise.all(orders.map(populateEntries));`.

But then this function would be more complicated.. not sure which is better..